### DISCUSSION TITLE
Idea for Relation Field

### DISCUSSION THREAD

**Discussion by hungcrush** [pocketbase/pocketbase#1735] [Q&A] [answered]
Currently when creating a record, the relation field is posted as ["e49vo9o76yovg49", ...]. 
What if they are objects? If they are objects, we will create a new record and insert the new insert ID into the relation field.

`["e49vo9o76yovg49"]` ==>  ``
[{ "name": "new record relation" }]
``

Additionally, if the object has an id, we will understand it as an update to the record.

What do you think about it?

The reason is I want to make repeated field.

**Comment by ganigeorgiev** [comment, marked as answer]
"Sub/Group/Repeatable" collections is in my todo to research it but for now is left aside since it will require too much internal changes and the use cases for it are not very clear to me.

Here are some things that you may have to consider if you are planning to work on it:

- note that a record can be created also with a custom id, so you can't rely only on the existence of the `id` field to detect whether it is update or create operation

- it may require to specify a path when loading the individual record form request data (_this is supported via the `RecordUpsert.LoadRequest(r, keyPrefix)`_)

- you'll have to wrap everything in one long running transaction, which depending on the value of allowed "Max select" may hinder other write requests because they will have to wait until the transaction manage to complete; nested file upload/delete will have to be also executed in a separate go routine(s) out of the main transaction to minimize the locking times (this is also one of the reason why transactional requests are not supported yet)

- it is ambiguous how the Create/Update API rules will be evaluated (will the rel/sub collection rules be ignored or there will be 2 different rules depending on whether is a direct create or create via inline relation submit, how the `@request.data.*` checks will look like, should the main model rel field literal be only the rel id or the serialized record object, etc.)

**Reply by hungcrush** [reply]
Well.. it's more complicated than I thought.. 
Thank you so much, I got it.
